### PR TITLE
feat(core): migrate breadcrumbs component

### LINKED ISSUE #8364: Enhancement: migrate breadcrumbs to use overflow layout
#### Is this a bug, enhancement, or feature request?
enhancement
#### Briefly describe your proposal.
Currently breadcrumbs component uses own implementation of overflow layout functionality.
This task is aimed on standartization of such components so they would use Overflow Layout component.

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #8364 

## Description
Migrated old breadcrumbs overflow mechanism to Overflow Layout component.

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/breadcrumb/breadcrumb.component.ts [hunk 1]
```diff
@@ -73,96 +57,72 @@ export class BreadcrumbComponent implements AfterViewInit, OnInit, OnDestroy {
...
-        firstValueFrom(this._ngZone.onStable).then(() => {
-            this._subscriptions.add(
-                this._resizeObserver
-                    .observe(this.containerElement || (this.elementRef.nativeElement.parentElement as Element))
-                    // Add small delay for performance reasons.
-                    .pipe(debounceTime(30))
-                    .subscribe(() => this.onResize())
-            );
-        });
+        this._rtlService?.rtl.subscribe((value) => this._placement$.next(value ? 'bottom-end' : 'bottom-start'));
```
Comment (mikerodonnell89) [inline libs/core/src/lib/breadcrumb/breadcrumb.component.ts:102]: Doesn't this now become a hanging subscription

**Review by mikerodonnell89 — COMMENTED** [review]
Now seeing some flickering when adding overflow items that wasn't there before:

https://user-images.githubusercontent.com/2471874/180234756-aea277d8-5746-44b3-9d8d-9c9ac57d56b6.mov

**Review by platon-rov — COMMENTED** [review]
as you refactored\changed overflow layout a lot in scope of that pr could you please fix unit tests related to it that failed?

**Review by platon-rov — APPROVED** [review]

**Comment by N1XUS** [comment]
> as you refactored\changed overflow layout a lot in scope of that pr could you please fix unit tests related to it that failed?

Checked. Tests related to Overflow Layout or Breadcrumbs are not causing any errors on local run.

**Review by g-cheishvili — APPROVED** [review]

**Pull request merged**